vernac/vernacentries.ml

```diff
@@ -1921,7 +1921,8 @@ let print_about_hyp_globs ~pstate ?loc ref_or_by_not udecl glopt =
       let open Constrexpr in
       match glnumopt, ref_or_by_not.v with
       | None,AN qid when qualid_is_ident qid -> (* goal number not given, catch any failure *)
-         (try get_nth_goal ~pstate 1, qualid_basename qid with _ -> raise NoHyp)
+         (try get_nth_goal ~pstate 1, qualid_basename qid
+          with e when CErrors.noncritical e -> raise NoHyp)
```

Comment:
```suggestion
          with Failure _ -> raise NoHyp)
```
like the one just below
or we could use nth_opt and return option in get_nth_goal

Reply:
I ended up getting rid of `get_nth_goal` entirely. The fact that it returns both a goal and sigma (that is independent of the goal) makes it a bit of an awkward helper function in my opinion.